mora/auth/auth.py

```diff
@@ -89,25 +104,17 @@ def login():
         raise exceptions.HTTPException(
             exceptions.ErrorCodes.E_CONNECTION_FAILED)
 
-    resp = flask.jsonify({
-        "user": username,
-    })
+    flask.session['username'] = username
 
     if assertion:
-        resp.set_cookie(
-            COOKIE_NAME, assertion,
-            secure=flask.request.is_secure,
-            httponly=True,
-        )
-    else:
-        resp.delete_cookie(COOKIE_NAME)
+        flask.session[COOKIE_NAME] = assertion
```

Comment:
Is it possible to hit the conditional above? In general, it seems to me that we should either assign `None` to the session key or clear it whenever authentication fails.